scripts/fota_builder.py

```diff
@@ -0,0 +1,363 @@
+        args = ["tar", "-C", rootfs_dir, "-xjf", conf["rootfs"]]
+        self.run_cmd(args)  # run tar
+
+        exclude = conf.get("exclude")
+        if exclude:
+            for item in exclude:
+                if item[0] == os.sep:
+                    item = item[1:]
+                dir_for_rm = os.path.join(rootfs_dir, item)
+                os.system(f"rm -rf {dir_for_rm}")
+
+        items = conf.get("items")
+        if items:
+            self.copy_files(items, rootfs_dir)
+
+        if overlay_type == "full":
+            self.overlay_full(metadata, rootfs_dir, bundle_dir)
+        elif overlay_type == "incremental":
+            self.overlay_incremental(metadata, rootfs_dir, bundle_dir)
+        else:
+            raise FotaError(EINVAL, f"Unknown bundle type {type}")
+        self.metadata["components"].append(metadata)
+
+    def do_custom_component(self, name, conf, bundle_dir):
+        """Create custom component"""
+        metadata = self.component_metadata(name, conf)
+        self.metadata["components"].append(metadata)
+        src = conf["file"]
+        self.do_copy(src, os.path.join(bundle_dir, metadata["fileName"]))
+
+    def process_component(self, name, conf, bundle_dir):
+        """Process component "name" """
+        method = conf["method"]
+        if method == "raw":
+            self.do_raw_component(name, conf, bundle_dir)
+        elif method == "overlay":
+            self.do_overlay_component(name, conf, bundle_dir)
+        elif method == "custom":
+            self.do_custom_component(name, conf, bundle_dir)
+
+    def run(self):
+        """Main class procedure"""
+        components = self.conf["components"]
+        work_dir = self.conf["work_dir"]
+        bundle_dir = os.path.join(work_dir, "bundle")
+        self.prepare_dir(bundle_dir)
+        for dom in components.keys():
+            print(f"Processing {dom}")
+            if not components[dom].get("enabled", True):
+                if self.verbose:
+                    print(f"{dom} skiped")
+                continue
+            self.process_component(dom, components[dom], bundle_dir)
+        manifest_file_name = os.path.join(bundle_dir, "manifest.json")
+        with open(manifest_file_name, "w", encoding="utf-8") as manifest_file:
+            json.dump(self.metadata, manifest_file, indent=4)
+        self.create_bundle()
+
+
+def main():
+    """main procedure"""
+    parser = argparse.ArgumentParser(description="fota")
+
+    parser.add_argument(
+        "conf", metavar="conf.yaml", type=str, help="YAML file with configuration"
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="store_true", help="Enable verbose output"
+    )
+
+    args = parser.parse_args()
+
+    with open(args.conf, "r", encoding="utf-8") as config_file:
+        conf = yaml.load(config_file, Loader=yaml.CLoader)
+    builder = FotaBuilder(conf, args.verbose)
+    try:
+        builder.run()
+    except FotaError as inst:
```

Comment:
`inst` stands for instance? Better to use such name as 'e', 'err', 'exc' in this case. 
https://docs.python.org/3/tutorial/errors.html